aws_lambda_powertools/utilities/data_classes/sqs_event.py

```diff
@@ -103,6 +103,16 @@ def body(self) -> str:
         """The message's contents (not URL-encoded)."""
         return self["body"]
 
+    @property
+    def json_body(self) -> Dict:
+        """Parses the submitted body as json"""
+        try:
+            if self._json_data is None:
+                self._json_data = self._json_deserializer(self["body"])
+            return self._json_data
+        except Exception:
+            return self["body"]
```

Comment:
The reason I add a `try/except` block here is because it could be a simple string that cannot be deserialized as `json`.
Should we consider this for the other `json_body` fields in Kinesis/Kafka/APIGW?

Reply:
We should fail hard here to give the caller a chance to handle this error correctly. 

Reasoning in a mental model fashion:

* Do we offer a non-JSON body property e.g., `body`?
    * If so, `json_body` should fail if it can't deserialize it
* If the customer received a poison pill record, a non-JSON string, will they be given the chance to handle a JSON deserialization error?
    * If we return `self["body"]`, it can result to a totally different error we can't even anticipate, e.g., `json_body["blah"]`

As a rule of thumb, you'd want to propagate the error as close to the actual problem so they instinctively know how to handle it. If we catch all, we're masking the actual error and leading them to an unknown error later on.

> tiny note: until Python 3.11, try/catch has a performance overhead even if it's tiny, so we can also improve here by letting them do it (not accidentally twice).